### PR TITLE
feat: Allow clients using matrix-sdk-ffi to set the user agent

### PR DESCRIPTION
**Pull request by Johennes**
Fixes: #1073

### CODE REVIEW

File: bindings/apple/Tests/MatrixRustSDKTests/ClientTests.swift [hunk 1]
```diff
@@ -11,6 +11,17 @@ final class ClientTests: XCTestCase {
             XCTFail("The client should build successfully when given a homeserver.")
         }
     }
+
+    func testBuildingWithHomeserverURLAndUserAgent() {
+        do {
+            _ = try ClientBuilder()
+                .homeserverUrl(url: "https://localhost:8008")
+                .userAgent(userAgent: "golden-eye/007")
```
[the remark this answers is not in the document]
Comment (Johennes) [reply bindings/apple/Tests/MatrixRustSDKTests/ClientTests.swift:19]: Hm, checking their [docs](https://mozilla.github.io/uniffi-rs/udl/functions.html) on functions, seems like there is nothing to allow overriding the argument label to `_`. 😕

**Review by jplatte — APPROVED** [review]

**Comment by jplatte** [comment]
Not sure why codecov doesn't run, can you rebase maybe?

File: bindings/apple/Tests/MatrixRustSDKTests/ClientTests.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (Johennes) [reply bindings/apple/Tests/MatrixRustSDKTests/ClientTests.swift:19]: Ah, yeah turning them off globally is probably even worse of an experience from the Swift side.

**Comment by Johennes** [comment]
> Not sure why codecov doesn't run, can you rebase maybe?

All green now.

**Pull request merged**